src/Databases/DatabaseOrdinary.cpp

```diff
@@ -215,8 +216,11 @@ LoadTaskPtr DatabaseOrdinary::startupTableAsync(
                 logAboutProgress(log, ++tables_started, total_tables_to_startup, startup_watch);
             }
             else
-                throw Exception(ErrorCodes::LOGICAL_ERROR, "Table {}.{} doesn't exist during startup",
+            {
+                // This might happen if synchronous metadata loading failed and server is going to shutdown.
+                throw Exception(ErrorCodes::UNKNOWN_TABLE, "Table {}.{} doesn't exist during startup",
```

Comment:
Okay, but do we have other places where a check like this is missing and we just assume that it never happens?

Reply:
IMO everything reasonable to do with a table should go after startup, so I think this one check should be good enough.